lib/EmitterInterface.php

```diff
@@ -74,5 +74,5 @@ public function removeListener(string $eventName, callable $listener): bool;
      * removed. If it is not specified, every listener for every event is
      * removed.
      */
-    public function removeAllListeners(string $eventName = null);
+    public function removeAllListeners(?string $eventName = null);
```

Comment:
For these kind of changes, if someone was implementing this interface, and has `string $eventName = null` and we change this "underneath them" adding the `?`, then will their code work unchanged?

LSP is not broken - the old and new interface, and the implementation of the interface all have exactly the same set of allowable things that can be passed in the parameter.

I wonder if this will break anything for anyone (e.g. in older versions of PHP) (callers will be fine - now it is explicit that they can pass `null`)

Reply:
I tested the following code in both PHP 7.4 and PHP 8.3 and it does not trigger any error.
```php
interface TestI {
    public function test(?string $a = null);
}
class Cls implements TestI {
    public function test(string $a = null) {}
}
```